app/payments/RazorpayPayment.php

```diff
@@ -0,0 +1,164 @@
+<?php
+use Razorpay\Api\Api;
+
+class RazorpayPayment
+{
+    public function getSessionValue($key)
+    {
+        return $_SESSION[$key];
+    }
+
+    public function setSessionValues($sessionValues)
+    {
+        foreach($sessionValues as $key => $value)
+        {
+            $_SESSION[$key] = $value;
+        }
+    }
+
+    public function getOrderData($orderId, $orderInfo, $processorData)
+    {
+        $data = array(
+            'receipt' => $orderId,
+            'amount' => fn_rzp_adjust_amount($orderInfo['total'], $processorData['processor_params']['currency'])*100,
+            'currency' => $processorData['processor_params']['currency'],
+            'payment_capture' => 1
+        );
+
+        return $data;
+    }
+
+    public function generateHtmlForm($url, $json)
+    {
+        $html = <<<EOT
+<script src="https://checkout.razorpay.com/v1/checkout.js"></script>
+<script>
+    var data = $json;
+    data.handler = function (transaction) {
+        document.getElementById('razorpay_payment_id').value = transaction.razorpay_payment_id;
+        document.getElementById('razorpay_signature').value = transaction.razorpay_signature;
+        document.getElementById('razorpay-form').submit();
+    };
+    function razorpaySubmit(){                  
+        var rzp1 = new Razorpay(data);
+        rzp1.open();
+        rzp1.modal.options.backdropClose = false;
+    }    
+            
+    var rzp_interval = setInterval(function(){
+        if (typeof window["Razorpay"] != "undefined")
+        {
+            setTimeout(function(){ razorpaySubmit(); }, 500);
```

Comment:
Explain this please? setInterval to check whether window['Razorpay'] is defined and then a setTimeout for when it is?

Can we not use document.onload or some other event instead? /cc @razorpay/frontend